### PR TITLE
MDS homework week three

### PR DESCRIPTION
**Pull request by AB-D**
-Added urls for clients and projects
-Updated -each- template to use {% url %} - tried to find a way to put this in the base.html template with an if statement comparing current url to another url (and failed)
-Created views that list clients and projects
-Used template inheritance
-Created client summary page that lists projects and time entires using
the pilfered table from ‘music_demo’

Thank you all. Apologies for the lack of comments but I don't think i did anything controversial.

### CODE REVIEW

**Comment by AB-D** [comment]
Hey, I don't think that i was clear about what I meant when i said - "Updated -each- template to use {% url %} - tried to find a way to put this in the base.html template with an if statement comparing current url to another url (and failed)"

I wanted to shift the navigation part into the base.html, but that would require some changes in order to tab "active"...

So, I thought about adding something like this for each navigation tab in the base.html (this is for the "entries" tab):
[...]
<ul class="nav">

```
    <li {% url 'entries' as entries_url %}{% if request.get_full_path == entries_url %}class="active"{% endif %}><a href="{% url 'entries' %}">Entries</a></li>
```

[...]

The navigation still worked when i did this, but the tab would not switch to "active". I was hunting around for guidance on this and there was some vague talk about the need to bring "request" in with the context, I tried to do that but with no success. Is something like this going to work? Or is there a better way to go about it?

I won't make it to class today, but will try and catch up from the slides for this week.

Cheers.

File: timetracker/entries/templates/clients.html [hunk 1]
```diff
@@ -0,0 +1,24 @@
+{% extends 'base.html' %}
+{% block title %}Projects{% endblock%}
+
+{% block content %}
+  
+  <body>
```
Comment (funkybob) [inline timetracker/entries/templates/clients.html:6]: You already have the `body` tag in the base template.

**Comment by sesh** [comment]
@AB-D Just a heads up – you forked the lesson-four repository a little early (you can fork it again now and it has the homework ready to complete from the class).

**Comment by AB-D** [comment]
@sesh thanks man. see you next week.

**Pull request closed without merging**